src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs

```diff
@@ -204,6 +204,24 @@ private async Task<DicomDataset> TryGetWorkitemBlobAsync(WorkitemInstanceIdentif
         }
     }
 
+    /// <inheritdoc />
+    public async Task<DicomDataset> RetrieveWorkitemAsync(string workitemInstanceUid, CancellationToken cancellationToken = default)
+    {
+        EnsureArg.IsNotEmptyOrWhiteSpace(workitemInstanceUid, nameof(workitemInstanceUid));
+
+        var partitionKey = _contextAccessor.RequestContext.GetPartitionKey();
+
+        var workitemMetadata = await _indexWorkitemStore
+            .GetWorkitemMetadataAsync(partitionKey, workitemInstanceUid, cancellationToken)
+            .ConfigureAwait(false);
+
+        var dataset = await _workitemStore
+            .GetWorkitemAsync(workitemMetadata, cancellationToken)
+            .ConfigureAwait(false);
+
+        return dataset;
```

Comment:
If the workitem moved to COMPLETED OR CANCELLED we need to send failure response.